// vite-plugin-inspect/dist/shared/vite-plugin-inspect.BzUKaD4x.mjs
<template>
  <div class="min-h-screen px-6 pt-32 max-w-6xl mx-auto">
    <div class="bg-white/90 backdrop-blur-sm rounded-xl p-4 shadow-lg mb-8 flex flex-wrap items-end gap-4 overflow-x-auto">
      <h2 class="text-2xl font-bold w-full md:w-auto mb-2 md:mb-0 flex-none">Recherche de vols :</h2>

      <!-- Départ -->
      <div class="flex-none relative">
        <label class="font-medium text-sm">Départ</label>
        <input type="text" v-model="search.departureName" @input="fetchDepartureSuggestions" placeholder="Ex: Paris" class="input-field" autocomplete="off"/>
        <ul v-if="filteredDeparture.length" class="autocomplete-list">
          <li v-for="(a, i) in filteredDeparture" :key="i" @click="selectDeparture(a)">
            {{ a.city }}, {{ a.country }}
          </li>
        </ul>
      </div>

      <!-- Arrivée -->
      <div class="flex-none relative">
        <label class="font-medium text-sm">Arrivée</label>
        <input type="text" v-model="search.arrivalName" @input="fetchArrivalSuggestions" placeholder="Ex: New York" class="input-field" autocomplete="off"/>
        <ul v-if="filteredArrival.length" class="autocomplete-list">
          <li v-for="(a, i) in filteredArrival" :key="i" @click="selectArrival(a)">
            {{ a.city }}, {{ a.country }}
          </li>
        </ul>
      </div>

      <!-- Date -->
      <div class="flex-none">
        <label class="font-medium text-sm">Date</label>
        <input v-model="search.date" type="date" class="input-field"/>
      </div>

      <div class="flex-none">
        <button @click="getFlights" class="bg-blue-500 hover:bg-blue-600 text-white font-semibold px-4 py-2 rounded shadow transition whitespace-nowrap">
          Chercher
        </button>
      </div>
    </div>

    <div v-if="loading" class="text-gray-700 font-medium mb-4">Chargement...</div>
    <div v-if="error" class="text-red-600 font-medium mb-4">{{ error }}</div>

    <div v-if="flights.length" class="grid gap-6">
      <div v-for="(f, i) in flights" :key="i" class="bg-white/90 backdrop-blur-sm rounded-xl p-4 shadow hover:shadow-2xl transition">
        <p><strong>{{ f.departure }} → {{ f.arrival }}</strong></p>
        <p>Compagnie: {{ f.airline }}</p>
        <p>Départ: {{ f.departure_time }}</p>
        <p>Arrivée: {{ f.arrival_time }}</p>
        <p>Prix: {{ f.price }} €</p>
      </div>
    </div>

    <div v-else-if="!loading" class="text-gray-500 font-medium">Aucun vol trouvé.</div>
  </div>
</template>

<script>
import axios from "axios";
import debounce from "lodash/debounce";

export default {
  data() {
    return {
      filteredDeparture: [],
      filteredArrival: [],
      flights: [],
      loading: false,
      error: "",
      search: {
        departureName: "",
        departure_id: "",
        arrivalName: "",
        arrival_id: "",
        date: "",
      },
    };
  },
  methods: {
    fetchDepartureSuggestions: debounce(async function() {
      if (!this.search.departureName) { this.filteredDeparture = []; return; }
      try {
        const res = await axios.get("http://localhost:3000/api/airports_suggestions", { params: { q: this.search.departureName } });
        this.filteredDeparture = res.data;
      } catch (err) { console.error(err); }
    }, 300),

    fetchArrivalSuggestions: debounce(async function() {
      if (!this.search.arrivalName) { this.filteredArrival = []; return; }
      try {
        const res = await axios.get("http://localhost:3000/api/airports_suggestions", { params: { q: this.search.arrivalName } });
        this.filteredArrival = res.data;
      } catch (err) { console.error(err); }
    }, 300),

    selectDeparture(a) {
      this.search.departureName = `${a.city}, ${a.country}`;
      this.search.departure_id = a.iata || a.id;
      this.filteredDeparture = [];
    },

    selectArrival(a) {
      this.search.arrivalName = `${a.city}, ${a.country}`;
      this.search.arrival_id = a.iata || a.id;
      this.filteredArrival = [];
    },

    async getFlights() {
      if (!this.search.departure_id || !this.search.arrival_id || !this.search.date) {
        this.error = "Veuillez remplir départ, arrivée et date.";
        return;
      }
      this.loading = true;
      this.error = "";
      this.flights = [];
      try {
        const res = await axios.get("http://localhost:3000/api/flights", {
          params: {
            departure_id: this.search.departure_id,
            arrival_id: this.search.arrival_id,
            date: this.search.date
          }
        });

        const flightResults = res.data.flight_results || [];
        this.flights = flightResults.map(f => ({
          departure: f.flights[0].departure_airport?.id,
          arrival: f.flights[f.flights.length-1].arrival_airport?.id,
          departure_time: f.flights[0].departure_airport?.time,
          arrival_time: f.flights[f.flights.length-1].arrival_airport?.time,
          airline: f.flights[0].airline || "Inconnu",
          price: f.price?.amount || 0
        }));

        if (!this.flights.length) this.error = "Aucun vol trouvé.";
      } catch (err) {
        console.error("Erreur getFlights :", err.response?.data || err.message);
        this.error = err.response?.data?.error || err.message;
      } finally {
        this.loading = false;
      }
    }
  }
};
</script>

<style scoped>
.input-field { min-width: 120px; padding: 0.4rem 0.6rem; border: 1px solid #ccc; border-radius: 0.5rem; }
.input-field:focus { outline: none; border-color: #8a4fff; box-shadow: 0 0 5px rgba(138,79,255,0.5); }
.relative { position: relative; }
.autocomplete-list { position: absolute; top: 100%; left: 0; right: 0; background: white; border: 1px solid #ccc; border-radius: 0.5rem; max-height: 200px; overflow-y: auto; z-index: 50; }
.autocomplete-list li { padding: 0.4rem 0.6rem; cursor: pointer; }
.autocomplete-list li:hover { background: #8a4fff22; }
</style>
